api/clientcredentialsloginprovider.go

```diff
@@ -20,18 +21,49 @@ var (
 	}
 )
 
+const (
+	// ClientIDEnvVar is the environment variable used to specify the client ID
+	// for client credentials authentication.
+	ClientIDEnvVar = "JUJU_CLIENT_ID"
```

Comment:
We have a loginprovider package. I wonder if these should go there.
Actually, shouldn't all the login provider implementation logic go there instead of this package?